File: BBIS.Api/ClientApp/src/models/TestOrder/UpdateTestOrderDto.ts
import { BloodTypingTestOrderDto } from './BloodTypingTestOrderDto';
import { BloodScreeningTestOrderDto } from './BloodScreeningTestOrderDto';
import { CoombsTestOrderDto } from './CoombsTestOrderDto';
import { UpdateCrossMatchTestOrderDto } from './UpdateCrossMatchTestOrderDto';

export class UpdateTestOrderDto {
    TestOrderId: Guid = '';
    PatientId: Guid | null = null;
    ReservationId: Guid | null = null;
    TestCompleted: boolean = false;
    Remarks: string = '';
    PerformedBy: Guid | null = null;
    ReviewedBy: Guid | null = null;
    ValidatedBy: Guid | null = null;
    PathologistId: Guid | null = null;
    CrossMatchTestOrders: Array<UpdateCrossMatchTestOrderDto> = [];
    CoombsTestOrder: CoombsTestOrderDto | null = null;
    BloodScreeningTestOrder: BloodScreeningTestOrderDto | null = null;
    BloodTypingTestOrder: BloodTypingTestOrderDto | null = null;
}

File: BBIS.Application/DTOs/TestOrder/UpdateTestOrderDto.cs
namespace BBIS.Application.DTOs.TestOrder
{
    public class UpdateTestOrderDto
    {
        public Guid TestOrderId { get; set; }
        public Guid PatientId { get; set; }
        public Guid ReservationId { get; set; }
        public bool TestCompleted { get; set; }
        public string Remarks { get; set; }
        public Guid PerformedBy { get; set; }
        public Guid ReviewedBy { get; set; }
        public Guid ValidatedBy { get; set; }
        public Guid PathologistId { get; set; }

        public CoombsTestOrderDto CoombsTestOrder { get; set; }
        public BloodScreeningTestOrderDto BloodScreeningTestOrder { get; set; }
        public BloodTypingTestOrderDto BloodTypingTestOrder { get; set; }

        public List<UpdateCrossMatchTestOrderDto> CrossMatchTestOrders { get; set; }
    }
}
